src/transaction/types/input.rs

```diff
@@ -488,6 +482,7 @@ impl Input {
             .chain(recipient)
             .chain(nonce.to_be_bytes())
             .chain(amount.to_be_bytes())
+            // TODO: Seems it is a bug and we need to pad `data` with zeros
```

Comment:
Oh, no, because here we do it without any padding=) https://github.com/FuelLabs/fuel-v2-contracts/blob/master/contracts/sidechain/FuelMessagePortal.sol#L130

Reply:
Hmm, but seems spec says that we need to do it with alignment
![Uploading image.png…]()